src/modules/wreck/lua.d/timeout.lua

```diff
@@ -0,0 +1,40 @@
+--
+-- Register timer on nodeid 0 if kvs `walltime` is set for this job.
+--  Kill job on timeout if reached.
+--
+
+local function start_timer (f, wreck, id, walltime)
+   local to, err = f:timer {
+        timeout = walltime*1000,
+        oneshot = true,
+        handler = function (f, to)
+          wreck:log_msg ("Timeout expired! Killing job")
+          local rc,err = f:sendevent ({signal = 14}, "wreck.%d.kill", id)
```

Comment:
We could, but that would require loading posix module which otherwise isn't needed.

I'm not tied to SIGALRM, but it mimics `run_timeout` and `alarm(3)` behavior, and also return code 142 can be checked for jobs terminated in this way, whereas SIGTERM may be used for jobs that cancelled?

Easy to change though.